### DISCUSSION TITLE
neo-tree as Telescope picker UI

### DISCUSSION THREAD

**Discussion by bennypowers** [nvim-neo-tree/neo-tree.nvim#116] [Q&A]
Hello! `NERDTree` => `nvim-tree.lua` => `nvim-neo-tree`, and I'm quite impressed even after a few minutes of use with the README's recommended config.

I would like to combine the best of `Telescope git_status` with `NeoTreeFloat git_status` - Would it be possible to make neo-tree the UI for the picker part of Telescope's git_status window?

**Comment by cseickel** [comment]
Anything is possible, but it might be more work to do that than to add the missing functionality to neo-tree itself.

I don't see any need for a fuzzy search over `git status` because it is usually a small enough list of files to fit on one screen. We already have a floating git status window. 

The only thing left that I am aware of is the ability to show the diff in a preview window. Is that the missing piece you are looking for?

**Reply by nyngwang** [reply]
Personally, I don't like telecope. It's laggy.

**Reply by bennypowers** [reply]
Still think a tree picker for Telescope would be cool, but yeah a preview window for `NeoTreeFloat git_status` would satisfy OP

**Reply by nyngwang** [reply]
@bennypowers: If you want to do git things I would happily recommend you to try [fzf-lua](https://github.com/ibhagwan/fzf-lua), where you can use `<Left>`, `<Right>` to do `git add` or undo. The related post is this one: https://www.reddit.com/r/neovim/comments/okewki/plugin_fzflua_fzf_plugin_written_in_pure_lua_with/

**Reply by amaksymov** [reply]
Hello, @cseickel! How can I get preview in `:Neotree float git_status`?

**Reply by cseickel** [reply]
@amaksymov There is no built-in way to do this yet, you would have to write the code to create that preview window and show it at the correct location. The trigger for showing that preview could be added as a custom command.

There is an open issue for adding preview functionality: #134

That issue also has some discussion on how to actually do it. If no one else picks it up, I will probably do it over the summer.

**Reply by nyngwang** [reply]
@amaksymov: FYI, the preview feature has been implemented by @cseickel . As per #134. But currently, the `:Neotree float git_status` command seems to have some falsy warnings.
